go/discovery_kit_sdk/discovery_http_adapter.go

```diff
@@ -91,3 +94,27 @@ func (a discoveryHttpAdapter) handleDiscover(w http.ResponseWriter, r *http.Requ
 	}
 	exthttp.WriteBody(w, body)
 }
+
+func (a discoveryHttpAdapter) checkForDuplicateTargets(targets []discovery_kit_api.Target) {
+	seenTargets := make(map[string]struct{})
+	for _, target := range targets {
+		key := target.Id + "|" + target.TargetType
```

Comment:
I'd go for a local declared struct for the key in this method, to avoid the string concatenation (and thus lower memory usage)